### DISCUSSION TITLE
Github App authentication weirdness

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [renovatebot/renovate#10109] [Request Help] [answered]
I'm trying to setup renovate on an enterprise github instance, where each repo owns their own renovate github action workflow (no centralized instance) so people have more control over how often it should run etc (and we have some huge repos that would just slow it down for smaller repos and so on).
There were some concerns with using personal access tokens for authenticating, since the scope is wide.
So I saw that renovate supports using tokens generated from a github app if you prefix it with `x-access-token:`
I tried that and was erroring out with:
```
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 60,
           "msg": "Authentication failure"
         }
       ]

```
and it is trying to access `https://{enterprise endpoint}/api/v3/user` which is not an endpoint that github apps are allowed to use on both enterprise and the public one
https://docs.github.com/en/rest/overview/endpoints-available-for-github-apps
https://docs.github.com/en/enterprise-server@3.1/rest/overview/endpoints-available-for-github-apps

So I looked in the code and saw there is this snippet that makes the call:
```
if (username) {
    renovateUsername = username;
  } else {
    userDetails = await getUserDetails(defaults.endpoint, token);
    renovateUsername = userDetails.username;
  }
```
So I tried setting `"username": "renovate"` in the `config.js` and that did indeed allow it to successfully run.
I checked the docs for `username` and see it's not even supposed to be used for anything other than bitbucket.
https://docs.renovatebot.com/self-hosted-configuration/#username
Is this the expected behavior?
Do we expect this workflow to keep working so I can rely on this github app setup with a dummy username in there?
Or is it important that the username is somehow valid?

**Comment by rarkins** [comment, marked as answer]
Please see https://github.com/renovatebot/renovate/issues/10040 and let me know if there's still any unanswered questions

**Reply by HenrikPoulsen** [reply]
Sounds perfect. Thanks :)
